samples/cpp/image_generation/text2image.cpp

```diff
@@ -1,27 +1,55 @@
 // Copyright (C) 2023-2025 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
+#include <iostream>
 
 #include "openvino/genai/image_generation/text2image_pipeline.hpp"
 
+
 #include "imwrite.hpp"
 #include "progress_bar.hpp"
 
 int32_t main(int32_t argc, char* argv[]) try {
-    OPENVINO_ASSERT(argc == 3, "Usage: ", argv[0], " <MODEL_DIR> '<PROMPT>'");
+    OPENVINO_ASSERT(argc == 3, "Usage: ", argv[0], " <MODEL_DIR> '<PROMPT>'");  // TODO: <PROMPT> unused
 
     const std::string models_path = argv[1], prompt = argv[2];
     const std::string device = "CPU";  // GPU can be used as well
 
+    std::vector<std::string> prompts = {
+        "happy dog",
+        "black cat",
+        "yellow raspberry",
+        "retro personal computer",
+        "walking astronaut",
+        "fish with a hat",
+        "flying car",
+    };
+
     ov::genai::Text2ImagePipeline pipe(models_path, device);
-    ov::Tensor image = pipe.generate(prompt,
-        ov::genai::width(512),
-        ov::genai::height(512),
-        ov::genai::num_inference_steps(20),
-        ov::genai::num_images_per_prompt(1),
-        ov::genai::callback(progress_bar));
-
-    // writes `num_images_per_prompt` images by pattern name
-    imwrite("image_%d.bmp", image, true);
+    
+    std::vector<std::thread> threads;
+
+    for (size_t i = 0; i < prompts.size(); ++i) {
+        const std::string p = prompts[i];
+        threads.emplace_back([i, &pipe, models_path, p] () {
+            std::cout << "Generating... " << i << std::endl;
+            ov::genai::Text2ImagePipeline::GenerationRequest request = pipe.create_generation_request();
+            request.set_scheduler(ov::genai::Scheduler::from_config(std::filesystem::path(models_path) / "scheduler/scheduler_config.json"));
```

Comment:
Please move this step to `clone()` methods

Reply:
Done. Saved root_dir for subsequent usage. Using root_dir we can re-create the scheduler during cloning.